wp-dependency-installer.php

```diff
@@ -604,6 +597,40 @@ public function get_config( $slug = '' ) {
 			return isset( $this->config[ $slug ] ) ? $this->config[ $slug ] : $this->config;
 		}
 
+		/**
+		 * Add Basic Auth headers for authentication.
+		 *
+		 * @param array  $args HTTP header args.
+		 * @param string $url  URL.
+		 *
+		 * @return array $args
+		 */
+		public function add_basic_auth_headers( $args, $url ) {
+			if ( null === $this->current_slug ) {
+				return $args;
+			}
+			$package = $this->config[ $this->current_slug ];
+			$host    = $package['host'];
+			$token   = empty( $package['token'] ) ? false : $package['token'];
+
+			if ( $token && $url === $package['download_link'] ) {
+				if ( 'github' === $host || 'gitea' === $host ) {
+					$args['headers']['Authorization'] = 'token ' . $token;
+				}
+				if ( 'gitlab' === $host ) {
+					$args['headers']['Authorization'] = 'Bearer ' . $token;
+				}
+			}
+
+			// dot org should not have auth header.
+			// phpcs:ignore WordPress.WP.CapitalPDangit.Misspelled
+			if ( 'wordpress' === $host ) {
+				unset( $args['headers']['Authorization'] );
```

Comment:
is this a known bug? (e.g. GHU)

Reply:
If GitHub Updater is running and the dependency is being updated by it then it can show the auth header. I was just cleaning things up here.